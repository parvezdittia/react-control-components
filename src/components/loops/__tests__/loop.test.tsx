import React from 'react';
import renderer from 'react-test-renderer';
import { Loop } from '..';

describe('<Loop />', () => {
  it('should render children for each item in the source array ', () => {
    const tree = renderer.create(
      <Loop source={[1, 2, 3]}>
        {(item, index) => <span key={index}>{item}</span>}
      </Loop>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render null when source array is empty', () => {
    const tree = renderer.create(
      <Loop source={[]}>
        {(item, index) => <span key={index}>{item}</span>}
      </Loop>
    );
    expect(tree).toMatchSnapshot();
  });
});
