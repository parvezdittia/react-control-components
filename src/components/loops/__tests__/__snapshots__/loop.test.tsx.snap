// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loop /> should render children for each item in the source array  1`] = `
[
  <span>
    1
  </span>,
  <span>
    2
  </span>,
  <span>
    3
  </span>,
]
`;

exports[`<Loop /> should render null when source array is empty 1`] = `null`;
